csrc/ir/utils.h

```diff
@@ -839,4 +840,41 @@ class TVDomainGuard {
 std::pair<std::vector<IterDomain*>, std::vector<IterDomain*>>
 getReshapeInputAndOutputIds(TensorView* reshape_out_tv);
 
+// Computes `Split`'s output contiguity. Returns the outer contiguity and then
+// the inner contiguity.
+std::pair<std::optional<bool>, std::optional<bool>> splitContiguity(
+    const std::optional<bool>& contiguity);
+
+// Computes `Merge`'s output contiguity. Returns a pair
+// `<mergeable,contiguity>`. `mergeable` indicates whether the two IterDomains
+// can be merged without materialization. For example, there's no way to merge
+// `outer=f,inner=t` while keeping the output as an alias, because a dimension
+// can only have one stride. `contiguity` is the contiguity of the merged output
+// IterDomain.
+//
+// Credits to @jacobhinkle:
+// https://github.com/NVIDIA/Fuser/pull/1124#discussion_r1368682735
+std::pair<bool, std::optional<bool>> mergeContiguity(
+    const bool outer_is_expanded,
+    const std::optional<bool>& outer_contiguity,
+    const bool inner_is_expanded,
+    const std::optional<bool>& inner_contiguity);
+
+// Given a TV, returns its layout with repsect to the logical domain. When
+// `allocation` is a split of `logical`, walks backwards from `allocation` to
+// `logical` to find a permutation of `logical` that satisfies the order in
+// `allocation`. The returned contiguity is computed according to
+// splitContiguity and mergeContiguity.
+//
+// Example:
+//   input TV:
+//     logical: [b, s, h]
+//     allocation: [s, d, h/d, b]
+//     contiguity: [t, t, f, t]
+//   output layout:
+//     allocation: [s, h, b]
+//     contiguity: [t, f, t]
+std::optional<LinkedHashMap<IterDomain*, std::optional<bool>>>
```

Comment:
```suggestion
Layout
```
I'd probably have a better answer if I knew how you plan to use this method. I'm worried that LinkedHashMap exposes too many implementation details to the user, causing this method to have to stick with that implementation.

Reply:
How about returning an optional pair of allocation domain and contiguity?
That will remove the dependence on using LinkedHashMap or any changes to Layout class?

Reply:
Thanks for reminding me that we have to return an optional because canonicalization may fail. I'd do std::optional<Layout>. Layout is meant to be a think wrapper of allocation_domain + contiguity, so I feel much more comfortable with exposing that than exposing LinkedHashMap. Optionally, you can move isCompliantWith out of the struct if you don't plan to use it, e.g., 
```
isCompliantWith(const Layout& layout, const Layout& required)
```
This function doesn't need accesses to private fields of Layout, so it was an overkill (by me) to make it a class method in the first place.